WallsAndHoles/tilemaphelpers.h

```diff
@@ -0,0 +1,37 @@
+#ifndef TILEMAPHELPERS_H
+#define TILEMAPHELPERS_H
+
+#include "tilemap.h"
+
+#include <QStack>
+#include <QSet>
+
+// QPoints are not hashable in Qt by default!
+inline uint qHash (const QPoint & key)
+{
+    return qHash (QPair<int,int>(key.x(), key.y()) );
+}
```

Comment:
I actually added this in array2dtools.h in my branch! Maybe we should put this into a separate header, and include it in both this header and my array2dtools.h header.

Reply:
Not a bad idea. It's so small though I'm not to worried if it were to just exist in both places, considering it is inline.

We can worry about it when merging your branch